ScrapeRepos.cjs

```diff
@@ -0,0 +1,243 @@
+      const repoName = $(el).text().trim();
+      repos.push(repoName);
+    });
+    
+    var trendingSplit = [];
+    // trim the repos to be correctly formatted
+    repos.forEach((repo) => {
+      let trimmedName = repo.replace(/\n\s+/g, '');
+      trimmedName = trimmedName.replace(/\//g, "");
+      const stringSplit = trimmedName.split(" ");
+      trendingSplit.push(stringSplit[0]);
+      trendingSplit.push(stringSplit[1]);
+    });
+
+    return trendingSplit;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**  This function imports the ReadMe.md file for a repository (if it can be located)
+* @param owner - owner of the repo
+* @param name - name of the repo
+* @returns a string containing the text of the repo
+*           returns null if the repo can't be located
+*/
+async function getReadme(owner, name) {
+  const readmePaths = [`https://raw.githubusercontent.com/${owner}/${name}/release/readme.md`,
+                         `https://raw.githubusercontent.com/${owner}/${name}/dev/README.rst`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/main/README.md`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/master/README.md`];
+ 
+  for(i = 0; i < readmePaths.length; i++){
+    try{
+      const response = await axios.get(readmePaths[i])
+      const content = response.data
+      const converter = new showdown.Converter();
+
+      // Use the converter object to convert Markdown to HTML to String:
+      const html = converter.makeHtml(content);
+      const html2 = html.toString();
+      const text = html2.replace(/<[^>]*>/g, '');
+      const text2 = text.replace(/\n\s+/g, '');
+      const text3 = text2.replace(/\//g, "");
+      return text3;
+    }
+    catch(error){
+
+    }
+  }
+  console.log("Couldn't locate the read me file")
+  return null
+}
+
+/** Gets the repo's information via GitHub's GraphQL API
+ * @param query GraphQL query for the repo (including owner and name)
+ * @param authToken personal authorization token
+ * @returns the json data for the requested repo as by the graphql query
+ */
+async function getRepoInfo(query, authToken){
+  try{
+    const response = await  axios.post('https://api.github.com/graphql', {
+      query: query
+    }, {
+      headers: {
+        'Authorization': authToken
+      }
+    });
+    const output = response.data.data.repository;
+    return output;
+  }
+  catch(error){
+    console.log(error);
+    return null
+  }
+}
+
+//TODO needs documentation, returns and revision!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! 
+/** Creates the ELI5 of the repository by passing the readme to chatgpt
+ * @param readME readMe file of the repository
+ * @returns SHOULD return the summary of the repo
+ */
+async function openAIRequestTurbo(readME) {
+  //const readme = await convertReadMe(); //here something wrong with the await
+  const OPENAI_API_KEY = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
```

Comment:
Key should not be stored here. This way is not secure. My suggestion is leaving like that for now and then creating a issue in linear to find the most secure way.

Reply:
Yes I totally agree with you.But for now (and testing purposes) we should leave it like that

Reply:
Yes I agree with you, please don't forget to open a new issue for this.

Reply:
I opened the issue on Linear and also added a discussion item to the sprint meeting